src/app/app-modules/nurse-doctor/case-sheet/general-case-sheet/doctor-diagnosis-case-sheet/doctor-diagnosis-case-sheet.component.ts

```diff
@@ -320,6 +324,66 @@ export class DoctorDiagnosisCaseSheetComponent
             );
         }
       }
+
+
+  
+      if (this.visitCategory === 'General OPD (QC)' && this.casesheetData && this.casesheetData.doctorData) {
+        this.referDetails = this.casesheetData.doctorData.Refer;
+        if (
+          this.referDetails &&
+          this.referDetails.refrredToAdditionalServiceList
+        ) {
+          console.log(
+            'institute',
+            this.referDetails.refrredToAdditionalServiceList,
+          );
+          for (
+            let i = 0;
+            i < this.referDetails.refrredToAdditionalServiceList.length;
+            i++
+          ) {
+            if (this.referDetails.refrredToAdditionalServiceList[i].serviceName) {
+              this.serviceList +=
+                this.referDetails.refrredToAdditionalServiceList[i].serviceName;
+              if (
+                i >= 0 && 
+                i < this.referDetails.refrredToAdditionalServiceList.length - 1
+              )
+                this.serviceList += ',';
+            }
+          }
+        }
+
+        if (this.referDetails && this.referDetails.referralReason) {
```

Comment:
`(this.referDetails && this.referDetails.referralReason)`
can be simplified to
`this.referDetails?.referralReason`

@devikasuresh20 read about https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining